swift/Ice/greeter/Sources/GreeterServer/main.swift

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) ZeroC, Inc.
+
+import Foundation
+import Ice
+
+// Create an Ice communicator to initialize the Ice runtime.
+let communicator = try Ice.initialize()
+
+// Destroy the communicator when the program exits.
+defer {
+    communicator.destroy()
+}
+
+// Set up a signal handler to catch Ctrl-C and shut down the communicator.
+signal(SIGINT) { _ in
```

Comment:
I'd rather not involve the communicator in the Ctrl+C handling, and do instead like in C++ and C#.